File: app/src/test/java/k/bs/designpatternsp/mediator/ja/MediatorTest.java
package k.bs.designpatternsp.mediator.ja;

import k.bs.designpatternsp.pattern.mediator.ja.ChatMediator;
import k.bs.designpatternsp.pattern.mediator.ja.ChatUser;

public class MediatorTest {
    public static void main(String[] args) {
        ChatMediator mediator = new ChatMediator();

        ChatUser john = new ChatUser(mediator,"John");

        mediator.addUser(new ChatUser(mediator,"Alice"))
                .addUser(new ChatUser(mediator,"Bob"))
                .addUser(john);

        john.send("Hi every one!");
    }
}

File: app/src/test/java/k/bs/designpatternsp/mediator/kt/MediatorTest.kt
package k.bs.designpatternsp.mediator.kt


import k.bs.designpatternsp.pattern.mediator.kt.ChatMediator
import k.bs.designpatternsp.pattern.mediator.kt.ChatUser
import org.junit.Test

class MediatorTest {

    @Test
    fun `Mediator`() {
        val mediator = ChatMediator()

        val john = ChatUser(mediator, "John")

        mediator
            .addUser(ChatUser(mediator, "Alice"))
            .addUser(ChatUser(mediator, "Bob"))
            .addUser(john)

        john.send("Hi everyone!")
    }
}